app/(candidate)/dashboard/content/components/MyContent.js

```diff
@@ -55,29 +70,146 @@ export default function MyContent({ campaign, prompts }) {
     return `${selected}21`;
   };
 
-  const mappedSections = Object.keys(sections).map((key) => {
-    return {
-      key,
-      title: camelToSentence(key),
-      icon: '/images/dashboard/slogan-icon.svg',
-    };
+  let inputData = [];
+  Object.keys(sections).forEach((key) => {
+    const section = sections[key];
+    inputData.push({
+      name: section.name,
+      updatedAt: section.updatedAt,
+      slug: camelToKebab(key),
+      documentKey: key,
+    });
   });
 
+  console.log('inputData', inputData);
+
+  const data = useMemo(() => inputData);
+
+  const columns = useMemo(() => [
+    {
+      Header: 'Name',
+      accessor: 'name',
+      Cell: ({ row }) => {
+        return (
+          <Link
+            href="/dashboard/content/[slug]"
+            as={`/dashboard/content/${row.original.slug}`}
+          >
+            <div className="flex flex-row items-center font-semibold">
+              <IoDocumentText className="ml-3" />
+              <div className="ml-3">{row.original.name}</div>
+            </div>
+          </Link>
+        );
+      },
+    },
+    {
+      Header: 'Last Modified',
+      accessor: 'updatedAt',
+      sortType: 'datetime',
+      Cell: ({ row }) => {
+        return (
+          <div className="pl-[40px]">
+            {dateUsHelper(row.original.updatedAt)}
+          </div>
+        );
+      },
+    },
+    {
+      Header: '',
+      collapse: true,
+      accessor: 'actions',
+      Cell: ({ row }) => {
+        const actionProps = {
+          slug: row.original?.slug ? row.original.slug : '',
+          name: row.original?.name ? row.original.name : '',
+          documentKey: row.original?.documentKey
+            ? row.original.documentKey
+            : '',
+          tableVersion,
+          updatedAt: row.original.updatedAt
+            ? row.original.updatedAt
+            : undefined,
+        };
+        return <Actions {...actionProps} />;
```

Comment:
this is ok, I would just pas {...row} and handle all the props in the Actions component